addons/mail/controllers/mail.py

```diff
@@ -165,3 +165,26 @@ def mail_action_view(self, model=None, res_id=None, access_token=None, **kwargs)
             except ValueError:
                 res_id = False
         return self._redirect_to_record(model, res_id, access_token, **kwargs)
+
+    @http.route('/mail/unfollow', type='http', auth='public')
+    def mail_action_unfollow(self, model, res_id, pid, token, **kwargs):
+        comparison, record, redirect = MailController._check_token_and_record_or_redirect(model, int(res_id), token)
+        if not comparison or not record:
+            raise AccessError(_('Non existing record or wrong token.'))
+
+        pid = int(pid)
```

Comment:
note that pid is not included in the token check... it means that you can use your own token from the record to unsubscribe all other partners from it, if you give the same token, but you hange the pid in the request (the check_access_right stuff will be ignored because the record is sudoed)

Reply:
Actually, it is if I am not wrong and the test check that tampering with the pid gives a 403 (“with self.subTest('Tampered partner id')”).

The route “/mail/unfollow” has pid as mandatory parameter and _check_token_and_record_or_redirect call _check_token that read params from the request and encode the parameters (and base link) to form the token and compare it to the one provided. So it must include the pid as well into the check (because it is a mandatory parameter).

Did I miss something ? (I just tested by changing the pid, got a 403, then not providing the pid got a 500 (TypeError: mail_action_unfollow() missing 1 required positional argument: 'pid').

Reply:
ah good catch, didn't notice, thanks for checking :)